### PR TITLE
PHP7.2 compatibility

### PR DESCRIPTION
**Pull request by RoyalPulp**
(affects v3)
each() and  create_function() are deprecated in PHP7.2 ,

-> I copied the code from the 4 branch - seems to work in my local environment

### CODE REVIEW

File: code/search/SearchForm.php [hunk 1]
```diff
@@ -173,23 +173,31 @@ public function getResults($pageLength = null, $data = null){
...
-		return implode(" ", $newWords);
-	}
+	protected function addStarsToKeywords($keywords)
+    {
+        if (!trim($keywords)) {
+            return "";
+        }
+        // Add * to each keyword
+        $splitWords = preg_split("/ +/", trim($keywords));
+        $newWords = [];
```
Comment (robbieaverill) [inline code/search/SearchForm.php]: We still need to support PHP 5.3, could you please use `array()` here?

**Comment by RoyalPulp** [comment]
hi robbie, if you know what has to be changed, could you you make the change? I am not really at my desk for 2 weeks - and I have then to look into, what has to be done to make it compatible...

**Comment by robbieaverill** [comment]
Thanks for your efforts =)

**Comment by robbieaverill** [comment]
Oops, we have a 3.7 patch branch for these kind of changes. I've cherry picked the commits back into 3.7

**Pull request merged**